### ISSUE TITLE
Add custom color palletes

### ISSUE THREAD

**Issue by AnyTimeTraveler** [guillaumechereau/goxel#105]
I like to work with my own set of colors and would like to be able to import my own color pallete.

**Comment by mrombout** [comment]
For those arriving from a search engine, custom color palettes are supported in Goxel 0.12.0 by copying a GIMP palettes to a specific location as described above.

1. Find the path by going to the menu `Edit > Settings`, then look under `Paths > Palettes`. E.g. on Linux it's `/home/<user>/.config/goxel/palettes`.
2. Copy your GIMP palette (a `.gpl` file) to that path, you may need to create the folders.
3. Restart Goxel
4. New palette is now available under `Palettes`.

Perhaps this issue can be closed?

**Comment by guillaumechereau** [comment]
I'll leave this open for now because it should either be documented somewhere in the app (settings maybe?) or even better we should have a way to create custom palette from goxel directly.  I'll try to work on that when I get the time.

**Comment by cowboy-bebug** [comment]
Yeah documentation would be great. Just for FYI, the path on macOS is set to be:
```
Library/Containers/com.noctuasoftware.goxel/Data/Library/Application\ Support/Goxel/palettes
```

It would be great to allow users to specify some env vars like:
```sh
# reference palette path from config file
export GOXEL_CONFIG="$HOME/.config/goxel/goxelrc"

# or reference palette path directly
export GOXEL_PALETTE_PATH="$HOME/.config/goxel/palettes"
```

Doing so will also make Goxel compliant with the [XDG spec](https://wiki.archlinux.org/title/XDG_Base_Directory) 😃
